src/frontend/src/pages/404.jsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import Header from '../components/Header';
+import Typography from '@material-ui/core/Typography';
+import { ThemeProvider, makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles((theme) => ({
+  h1: {
+    position: 'absolute',
+    color: 'white',
+    fontFamily: 'Roboto',
+    fontWeight: 'bold',
+    opacity: 0.85,
+    fontSize: '12rem',
+    display: 'block',
+    top: theme.spacing(20),
+    left: theme.spacing(8),
+    [theme.breakpoints.between('xs', 'sm')]: {
+      fontSize: '4rem',
+      textAlign: 'left',
+      left: theme.spacing(4),
+      right: theme.spacing(4),
+      top: theme.spacing(14),
+    },
+    [theme.breakpoints.between('md', 'lg')]: {
+      fontSize: '8rem',
+    },
+    [theme.breakpoints.up('xl')]: {
+      fontSize: '12rem',
+    },
+  },
+  h2: {
+    position: 'absolute',
+    color: 'white',
+    fontFamily: 'Roboto',
+    fontSize: '2rem',
+    display: 'block',
+    bottom: theme.spacing(12),
+    left: theme.spacing(8),
+    lineHeight: 'inherit',
+    letterSpacing: 'inherit',
+    transition: 'all linear 1s',
+    [theme.breakpoints.between('xs', 'sm')]: {
+      textAlign: 'left',
+      fontSize: '2rem',
+      left: theme.spacing(4),
+      right: theme.spacing(4),
+    },
+    [theme.breakpoints.between('md', 'lg')]: {
+      fontSize: '4rem',
+    },
+    [theme.breakpoints.up('xl')]: {
+      fontSize: '8rem',
+    },
+  },
+}));
+
+const ErrorPage = (props) => {
+  const classes = useStyles();
+  const params = new URLSearchParams(props.location.search);
+  let originalUrl = params
+    .get('search')
+    ?.replace('/', '')
```

Comment:
I don't understand what this is doing.  I'm also not sure that we should introduce `?.` to our code base without discussing implications.

Reply:
The ?. is the new optional chaining operator, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

It's essentially a replacement of the following semantic: 

``` javascript
const a= params.get('search');
 if (a) {
    const b = a.replace('/', '');

   if (b) {
     const c = b.match(....);

     ... and so on
   }
 }
```

The reason for the actual functions is removing the `/` at the start of the URL search param, which I may be able to remove with the urlEncode that you suggested above. Furthermore, it formats the param client-side to `Capital Spaced` if capital letters exist. Can remove, thought it was a nice little addon.